src/createEpicMiddleware.ts

```diff
@@ -1,57 +1,52 @@
-import { Action, Middleware, MiddlewareAPI, Dispatch } from 'redux';
-import { Subject, from, queueScheduler } from 'rxjs';
+import { Action as ReduxAction, Dispatch, Middleware, MiddlewareAPI } from 'redux';
+import { from, Subject } from 'rxjs';
+import { QueueAction } from 'rxjs/internal/scheduler/QueueAction';
+import { QueueScheduler } from 'rxjs/internal/scheduler/QueueScheduler';
 import { map, mergeMap, observeOn, subscribeOn } from 'rxjs/operators';
-import { StateObservable } from './StateObservable';
 import { Epic } from './epic';
+import { StateObservable } from './StateObservable';
 import { warn } from './utils/console';
 
-interface Options<D = any> {
-  dependencies?: D;
+interface Options<Dependencies = any> {
+  dependencies?: Dependencies;
 }
 
 export interface EpicMiddleware<
-  T extends Action,
-  O extends T = T,
-  S = any,
-  D = any
-> extends Middleware<{}, S, Dispatch<any>> {
-  run(rootEpic: Epic<T, O, S, D>): void;
+  Action extends ReduxAction,
+  Output extends Action = Action,
+  State = any,
+  Dependencies = any
+> extends Middleware<{}, State, Dispatch<any>> {
+  run(rootEpic: Epic<Action, Output, State, Dependencies>): void;
 }
 
 export function createEpicMiddleware<
-  T extends Action,
-  O extends T = T,
-  S = any,
-  D = any
->(options: Options<D> = {}): EpicMiddleware<T, O, S, D> {
-  // This isn't great. RxJS doesn't publicly export the constructor for
-  // QueueScheduler nor QueueAction, so we reach in. We need to do this because
-  // we don't want our internal queuing mechanism to be on the same queue as any
-  // other RxJS code outside of redux-observable internals.
-  const QueueScheduler: any = queueScheduler.constructor;
-  const uniqueQueueScheduler: typeof queueScheduler = new QueueScheduler(
-    (queueScheduler as any).SchedulerAction
-  );
```

Comment:
We didn’t use an internal import because those imports are not guaranteed to be stable between minor/patch releases because they are internal. Whereas, while dirty, the `queueScheduler.constructor` technique should always work as long as queueScheduler is exported, which if changed wouldn’t until a major version release.

Reply:
The real solution is to have rxjs export the constructor publicly, but we just haven't got around to it haha

Reply:
Ah, okay then. I'll revert this and perhaps reference this PR in case of future PRs.